Does not build on OSX

The README file states that this works on OSX, but the current version doesn't compile. Here is the output from make:

~~~~
mkdir -p obj
mkdir -p obj/transport
mkdir -p obj/util
mkdir -p obj/crypto
mkdir -p obj/tunnel
mkdir -p obj/i2pcontrol
clang++ -g -Wall -std=c++11 -DCRYPTOPP_DISABLE_ASM -DMAC_OSX -maes -DAESNI  -I/usr/local/include -I.  -c -o obj/transport/NTCPSession.o transport/NTCPSession.cpp
In file included from transport/NTCPSession.cpp:10:
In file included from ./I2NPProtocol.h:11:
In file included from ./RouterInfo.h:9:
In file included from /usr/local/include/boost/asio.hpp:21:
In file included from /usr/local/include/boost/asio/basic_datagram_socket.hpp:20:
In file included from /usr/local/include/boost/asio/basic_socket.hpp:20:
In file included from /usr/local/include/boost/asio/basic_io_object.hpp:19:
In file included from /usr/local/include/boost/asio/io_service.hpp:24:
In file included from /usr/local/include/boost/asio/detail/wrapped_handler.hpp:18:
In file included from /usr/local/include/boost/asio/detail/bind_handler.hpp:19:
In file included from /usr/local/include/boost/asio/detail/handler_alloc_helpers.hpp:21:
In file included from /usr/local/include/boost/asio/handler_alloc_hook.hpp:80:
In file included from /usr/local/include/boost/asio/impl/handler_alloc_hook.ipp:19:
In file included from /usr/local/include/boost/asio/detail/call_stack.hpp:20:
In file included from /usr/local/include/boost/asio/detail/tss_ptr.hpp:27:
In file included from /usr/local/include/boost/asio/detail/posix_tss_ptr.hpp:76:
In file included from /usr/local/include/boost/asio/detail/impl/posix_tss_ptr.ipp:24:
In file included from /usr/local/include/boost/asio/error.hpp:28:
In file included from ./netdb.h:16:
In file included from ./tunnel/Tunnel.h:13:
In file included from ./tunnel/TunnelConfig.h:10:
In file included from ./RouterContext.h:13:
./Garlic.h:83:29: error: use of undeclared identifier 'I2NPMessage'
            std::shared_ptr<I2NPMessage> WrapSingleMessage (std::shared_ptr<const I2NPMessage> msg);
                            ^
./Garlic.h:83:83: error: unknown type name 'I2NPMessage'
            std::shared_ptr<I2NPMessage> WrapSingleMessage (std::shared_ptr<const I2NPMessage> msg);
                                                                                  ^
./Garlic.h:94:73: error: unknown type name 'I2NPMessage'
            size_t CreateAESBlock (uint8_t * buf, std::shared_ptr<const I2NPMessage> msg);
                                                                        ^
./Garlic.h:95:82: error: unknown type name 'I2NPMessage'
            size_t CreateGarlicPayload (uint8_t * payload, std::shared_ptr<const I2NPMessage> msg, UnconfirmedTags * newTags);
                                                                                 ^
./Garlic.h:96:76: error: unknown type name 'I2NPMessage'
            size_t CreateGarlicClove (uint8_t * buf, std::shared_ptr<const I2NPMessage> msg, bool isDestination);
                                                                           ^
./Garlic.h:129:29: error: use of undeclared identifier 'I2NPMessage'
            std::shared_ptr<I2NPMessage> WrapMessage (std::shared_ptr<const i2p::data::RoutingDestination> destination, 
                            ^
./Garlic.h:130:33: error: use of undeclared identifier 'I2NPMessage'
                std::shared_ptr<I2NPMessage> msg, bool attachLeaseSet = false);
                                ^
./Garlic.h:136:64: error: use of undeclared identifier 'I2NPMessage'
            virtual void ProcessGarlicMessage (std::shared_ptr<I2NPMessage> msg);
                                                               ^
./Garlic.h:137:72: error: use of undeclared identifier 'I2NPMessage'
            virtual void ProcessDeliveryStatusMessage (std::shared_ptr<I2NPMessage> msg);           
                                                                       ^
./Garlic.h:142:107: error: no member named 'InboundTunnel' in namespace 'i2p::tunnel'
            virtual void HandleI2NPMessage (const uint8_t * buf, size_t len, std::shared_ptr<i2p::tunnel::InboundTunnel> from) = 0;
                                                                                             ~~~~~~~~~~~~~^
./Garlic.h:146:55: error: use of undeclared identifier 'I2NPMessage'
            void HandleGarlicMessage (std::shared_ptr<I2NPMessage> msg);
                                                      ^
./Garlic.h:147:63: error: use of undeclared identifier 'I2NPMessage'
            void HandleDeliveryStatusMessage (std::shared_ptr<I2NPMessage> msg);            
                                                              ^
./Garlic.h:152:46: error: no member named 'InboundTunnel' in namespace 'i2p::tunnel'
                std::shared_ptr<i2p::tunnel::InboundTunnel> from);
                                ~~~~~~~~~~~~~^
./Garlic.h:153:95: error: no member named 'InboundTunnel' in namespace 'i2p::tunnel'
            void HandleGarlicPayload (uint8_t * buf, size_t len, std::shared_ptr<i2p::tunnel::InboundTunnel> from);
                                                                                 ~~~~~~~~~~~~~^
In file included from transport/NTCPSession.cpp:10:
In file included from ./I2NPProtocol.h:11:
In file included from ./RouterInfo.h:9:
In file included from /usr/local/include/boost/asio.hpp:21:
In file included from /usr/local/include/boost/asio/basic_datagram_socket.hpp:20:
In file included from /usr/local/include/boost/asio/basic_socket.hpp:20:
In file included from /usr/local/include/boost/asio/basic_io_object.hpp:19:
In file included from /usr/local/include/boost/asio/io_service.hpp:24:
In file included from /usr/local/include/boost/asio/detail/wrapped_handler.hpp:18:
In file included from /usr/local/include/boost/asio/detail/bind_handler.hpp:19:
In file included from /usr/local/include/boost/asio/detail/handler_alloc_helpers.hpp:21:
In file included from /usr/local/include/boost/asio/handler_alloc_hook.hpp:80:
In file included from /usr/local/include/boost/asio/impl/handler_alloc_hook.ipp:19:
In file included from /usr/local/include/boost/asio/detail/call_stack.hpp:20:
In file included from /usr/local/include/boost/asio/detail/tss_ptr.hpp:27:
In file included from /usr/local/include/boost/asio/detail/posix_tss_ptr.hpp:76:
In file included from /usr/local/include/boost/asio/detail/impl/posix_tss_ptr.ipp:24:
In file included from /usr/local/include/boost/asio/error.hpp:28:
In file included from ./netdb.h:16:
In file included from ./tunnel/Tunnel.h:13:
In file included from ./tunnel/TunnelConfig.h:10:
./RouterContext.h:38:24: error: no type named 'RouterInfo' in namespace 'i2p::data'
            i2p::data::RouterInfo& GetRouterInfo () { return m_RouterInfo; };
            ~~~~~~~~~~~^
./RouterContext.h:39:46: error: no type named 'RouterInfo' in namespace 'i2p::data'
            std::shared_ptr<const i2p::data::RouterInfo> GetSharedRouterInfo () const 
                                  ~~~~~~~~~~~^
./RouterContext.h:52:52: error: no member named 'ip' in namespace 'boost::asio'
            void UpdateAddress (const boost::asio::ip::address& host);  // called from SSU or Daemon
                                      ~~~~~~~~~~~~~^
./RouterContext.h:53:50: error: no type named 'RouterInfo' in namespace 'i2p::data'
            bool AddIntroducer (const i2p::data::RouterInfo& routerInfo, uint32_t tag);
                                      ~~~~~~~~~~~^
./RouterContext.h:54:55: error: no member named 'ip' in namespace 'boost::asio'
            void RemoveIntroducer (const boost::asio::ip::udp::endpoint& e);
                                         ~~~~~~~~~~~~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
~~~~

I'm using Boost 1.58.0 (installed via brew).

clang is the default version that comes with osx 10.10.4

~~~~
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.4.0
Thread model: posix
~~~

and cmake gives this error:

    CMake Error: The source directory "/usr/local/i2pd" does not appear to contain CMakeLists.txt.

Ok. I used cmake in build, which generated the build files. Then I used make in the main directory which gives the following errors:

~~~~
[  2%] Building CXX object CMakeFiles/common.dir/transport/NTCPSession.cpp.o
In file included from /usr/local/i2pd/transport/NTCPSession.cpp:10:
In file included from /usr/local/i2pd/build/../I2NPProtocol.h:10:
In file included from /usr/local/i2pd/build/../Identity.h:10:
/usr/local/i2pd/build/../crypto/Signature.h:437:47: warning: unused parameter 'signingPrivateKey' [-Wunused-parameter]
            EDDSA25519Signer (const uint8_t * signingPrivateKey) {}; 
                                              ^
In file included from /usr/local/i2pd/transport/NTCPSession.cpp:10:
In file included from /usr/local/i2pd/build/../I2NPProtocol.h:11:
In file included from /usr/local/i2pd/build/../RouterInfo.h:9:
In file included from /usr/local/include/boost/asio.hpp:21:
In file included from /usr/local/include/boost/asio/basic_datagram_socket.hpp:20:
In file included from /usr/local/include/boost/asio/basic_socket.hpp:20:
In file included from /usr/local/include/boost/asio/basic_io_object.hpp:19:
In file included from /usr/local/include/boost/asio/io_service.hpp:24:
In file included from /usr/local/include/boost/asio/detail/wrapped_handler.hpp:18:
In file included from /usr/local/include/boost/asio/detail/bind_handler.hpp:19:
In file included from /usr/local/include/boost/asio/detail/handler_alloc_helpers.hpp:21:
In file included from /usr/local/include/boost/asio/handler_alloc_hook.hpp:80:
In file included from /usr/local/include/boost/asio/impl/handler_alloc_hook.ipp:19:
In file included from /usr/local/include/boost/asio/detail/call_stack.hpp:20:
In file included from /usr/local/include/boost/asio/detail/tss_ptr.hpp:27:
In file included from /usr/local/include/boost/asio/detail/posix_tss_ptr.hpp:76:
In file included from /usr/local/include/boost/asio/detail/impl/posix_tss_ptr.ipp:24:
In file included from /usr/local/include/boost/asio/error.hpp:28:
In file included from /usr/local/i2pd/build/../netdb.h:16:
In file included from /usr/local/i2pd/build/../tunnel/Tunnel.h:13:
In file included from /usr/local/i2pd/build/../tunnel/TunnelConfig.h:10:
In file included from /usr/local/i2pd/build/../RouterContext.h:13:
/usr/local/i2pd/build/../Garlic.h:83:29: error: use of undeclared identifier 'I2NPMessage'
            std::shared_ptr<I2NPMessage> WrapSingleMessage (std::shared_ptr<const I2NPMessage> msg);
                            ^
/usr/local/i2pd/build/../Garlic.h:83:83: error: unknown type name 'I2NPMessage'
            std::shared_ptr<I2NPMessage> WrapSingleMessage (std::shared_ptr<const I2NPMessage> msg);
                                                                                  ^
/usr/local/i2pd/build/../Garlic.h:94:73: error: unknown type name 'I2NPMessage'
            size_t CreateAESBlock (uint8_t * buf, std::shared_ptr<const I2NPMessage> msg);
                                                                        ^
/usr/local/i2pd/build/../Garlic.h:95:82: error: unknown type name 'I2NPMessage'
            size_t CreateGarlicPayload (uint8_t * payload, std::shared_ptr<const I2NPMessage> msg, UnconfirmedTags * newTags);
                                                                                 ^
/usr/local/i2pd/build/../Garlic.h:96:76: error: unknown type name 'I2NPMessage'
            size_t CreateGarlicClove (uint8_t * buf, std::shared_ptr<const I2NPMessage> msg, bool isDestination);
                                                                           ^
/usr/local/i2pd/build/../Garlic.h:129:29: error: use of undeclared identifier 'I2NPMessage'
            std::shared_ptr<I2NPMessage> WrapMessage (std::shared_ptr<const i2p::data::RoutingDestination> destination, 
                            ^
/usr/local/i2pd/build/../Garlic.h:130:33: error: use of undeclared identifier 'I2NPMessage'
                std::shared_ptr<I2NPMessage> msg, bool attachLeaseSet = false);
                                ^
/usr/local/i2pd/build/../Garlic.h:136:64: error: use of undeclared identifier 'I2NPMessage'
            virtual void ProcessGarlicMessage (std::shared_ptr<I2NPMessage> msg);
                                                               ^
/usr/local/i2pd/build/../Garlic.h:137:72: error: use of undeclared identifier 'I2NPMessage'
            virtual void ProcessDeliveryStatusMessage (std::shared_ptr<I2NPMessage> msg);           
                                                                       ^
/usr/local/i2pd/build/../Garlic.h:142:107: error: no member named 'InboundTunnel' in namespace 'i2p::tunnel'
            virtual void HandleI2NPMessage (const uint8_t * buf, size_t len, std::shared_ptr<i2p::tunnel::InboundTunnel> from) = 0;
                                                                                             ~~~~~~~~~~~~~^
/usr/local/i2pd/build/../Garlic.h:146:55: error: use of undeclared identifier 'I2NPMessage'
            void HandleGarlicMessage (std::shared_ptr<I2NPMessage> msg);
                                                      ^
/usr/local/i2pd/build/../Garlic.h:147:63: error: use of undeclared identifier 'I2NPMessage'
            void HandleDeliveryStatusMessage (std::shared_ptr<I2NPMessage> msg);            
                                                              ^
/usr/local/i2pd/build/../Garlic.h:152:46: error: no member named 'InboundTunnel' in namespace 'i2p::tunnel'
                std::shared_ptr<i2p::tunnel::InboundTunnel> from);
                                ~~~~~~~~~~~~~^
/usr/local/i2pd/build/../Garlic.h:153:95: error: no member named 'InboundTunnel' in namespace 'i2p::tunnel'
            void HandleGarlicPayload (uint8_t * buf, size_t len, std::shared_ptr<i2p::tunnel::InboundTunnel> from);
                                                                                 ~~~~~~~~~~~~~^
In file included from /usr/local/i2pd/transport/NTCPSession.cpp:10:
In file included from /usr/local/i2pd/build/../I2NPProtocol.h:11:
In file included from /usr/local/i2pd/build/../RouterInfo.h:9:
In file included from /usr/local/include/boost/asio.hpp:21:
In file included from /usr/local/include/boost/asio/basic_datagram_socket.hpp:20:
In file included from /usr/local/include/boost/asio/basic_socket.hpp:20:
In file included from /usr/local/include/boost/asio/basic_io_object.hpp:19:
In file included from /usr/local/include/boost/asio/io_service.hpp:24:
In file included from /usr/local/include/boost/asio/detail/wrapped_handler.hpp:18:
In file included from /usr/local/include/boost/asio/detail/bind_handler.hpp:19:
In file included from /usr/local/include/boost/asio/detail/handler_alloc_helpers.hpp:21:
In file included from /usr/local/include/boost/asio/handler_alloc_hook.hpp:80:
In file included from /usr/local/include/boost/asio/impl/handler_alloc_hook.ipp:19:
In file included from /usr/local/include/boost/asio/detail/call_stack.hpp:20:
In file included from /usr/local/include/boost/asio/detail/tss_ptr.hpp:27:
In file included from /usr/local/include/boost/asio/detail/posix_tss_ptr.hpp:76:
In file included from /usr/local/include/boost/asio/detail/impl/posix_tss_ptr.ipp:24:
In file included from /usr/local/include/boost/asio/error.hpp:28:
In file included from /usr/local/i2pd/build/../netdb.h:16:
In file included from /usr/local/i2pd/build/../tunnel/Tunnel.h:13:
In file included from /usr/local/i2pd/build/../tunnel/TunnelConfig.h:10:
/usr/local/i2pd/build/../RouterContext.h:38:24: error: no type named 'RouterInfo' in namespace 'i2p::data'
            i2p::data::RouterInfo& GetRouterInfo () { return m_RouterInfo; };
            ~~~~~~~~~~~^
/usr/local/i2pd/build/../RouterContext.h:39:46: error: no type named 'RouterInfo' in namespace 'i2p::data'
            std::shared_ptr<const i2p::data::RouterInfo> GetSharedRouterInfo () const 
                                  ~~~~~~~~~~~^
/usr/local/i2pd/build/../RouterContext.h:52:52: error: no member named 'ip' in namespace 'boost::asio'
            void UpdateAddress (const boost::asio::ip::address& host);  // called from SSU or Daemon
                                      ~~~~~~~~~~~~~^
/usr/local/i2pd/build/../RouterContext.h:53:50: error: no type named 'RouterInfo' in namespace 'i2p::data'
            bool AddIntroducer (const i2p::data::RouterInfo& routerInfo, uint32_t tag);
                                      ~~~~~~~~~~~^
/usr/local/i2pd/build/../RouterContext.h:54:55: error: no member named 'ip' in namespace 'boost::asio'
            void RemoveIntroducer (const boost::asio::ip::udp::endpoint& e);
                                         ~~~~~~~~~~~~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
1 warning and 20 errors generated.
make[2]: *** [CMakeFiles/common.dir/transport/NTCPSession.cpp.o] Error 1
make[1]: *** [CMakeFiles/common.dir/all] Error 2
~~~~
If I'm doing something wrong, let me know. I never use cmake.

@EinMByte Can confirm I'm seeing the same on OS X with your branch:

    In file included from /Users/ric/Desktop/working/i2pd/core/RouterContext.h:13:
    /Users/ric/Desktop/working/i2pd/core/Garlic.h:83:29: error: use of undeclared identifier 'I2NPMessage'
                std::shared_ptr<I2NPMessage> WrapSingleMessage (std::shared_ptr<const I2NPMessage> msg);
                                ^

etc.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,7 +60,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
         )
     endif()
 elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    # TODO: more tweaks
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -undefined dynamic_lookup")
 endif()
 
 # Compiler flags customization (by system)

--- a/client/Daemon.cpp
+++ b/client/Daemon.cpp
@@ -9,7 +9,7 @@
 #include "RouterInfo.h"
 #include "RouterContext.h"
 #include "tunnel/Tunnel.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "Garlic.h"
 #include "util/util.h"
 #include "Streaming.h"

--- a/client/HTTPServer.cpp
+++ b/client/HTTPServer.cpp
@@ -7,7 +7,7 @@
 #include "tunnel/Tunnel.h"
 #include "tunnel/TransitTunnel.h"
 #include "transport/Transports.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "util/I2PEndian.h"
 #include "Streaming.h"
 #include "Destination.h"

--- a/client/i2pcontrol/I2PControl.cpp
+++ b/client/i2pcontrol/I2PControl.cpp
@@ -15,7 +15,7 @@
 #include "util/Timestamp.h"
 #include "transport/Transports.h"
 #include "tunnel/Tunnel.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "version.h"
 #include "Daemon.h"
 

--- a/core/AddressBook.cpp
+++ b/core/AddressBook.cpp
@@ -12,7 +12,7 @@
 #include "util/util.h"
 #include "Identity.h"
 #include "util/Log.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "AddressBook.h"
 #include "Destination.h"
 

--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -22,7 +22,7 @@ set(CORE_SRC
   "Identity.cpp"
   "LeaseSet.cpp"
   "NetDbRequests.cpp"	
-  "NetDb.cpp"
+  "NetworkDatabase.cpp"
   "Profiling.cpp"
   "RouterContext.cpp"
   "RouterInfo.cpp"

--- a/core/Destination.cpp
+++ b/core/Destination.cpp
@@ -5,7 +5,7 @@
 #include "util/util.h"
 #include "crypto/ElGamal.h"
 #include "util/Timestamp.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "Destination.h"
 #include "AddressBook.h"
 

--- a/core/Destination.h
+++ b/core/Destination.h
@@ -14,7 +14,7 @@
 #include "crypto/CryptoConst.h"
 #include "LeaseSet.h"
 #include "Garlic.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "Streaming.h"
 #include "Datagram.h"
 

--- a/core/I2NPProtocol.cpp
+++ b/core/I2NPProtocol.cpp
@@ -5,7 +5,7 @@
 #include "crypto/ElGamal.h"
 #include "util/Timestamp.h"
 #include "RouterContext.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "tunnel/Tunnel.h"
 #include "transport/Transports.h"
 #include "Garlic.h"

--- a/core/LeaseSet.cpp
+++ b/core/LeaseSet.cpp
@@ -5,7 +5,7 @@
 #include "crypto/CryptoConst.h"
 #include "util/Log.h"
 #include "util/Timestamp.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "tunnel/TunnelPool.h"
 #include "LeaseSet.h"
 

--- a/core/NetDbRequests.cpp
+++ b/core/NetDbRequests.cpp
@@ -1,7 +1,7 @@
 #include "util/Log.h"
 #include "I2NPProtocol.h"
 #include "transport/Transports.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "NetDbRequests.h"
 
 namespace i2p

--- a/core/NetworkDatabase.cpp
+++ b/core/NetworkDatabase.cpp
@@ -12,7 +12,7 @@
 #include "transport/Transports.h"
 #include "RouterContext.h"
 #include "Garlic.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "util/util.h"
 
 using namespace i2p::transport;

--- a/core/NetworkDatabase.h
+++ b/core/NetworkDatabase.h
@@ -1,3 +1,8 @@
+/**
+ * @note This file is named NetworkDatabase.h instead of NetDb.h to avoid
+ *  a possible nameclash with a file included by boost on case insensitive
+ *  filesystems.
+ */
 #ifndef NETDB_H__
 #define NETDB_H__
 

--- a/core/Reseed.cpp
+++ b/core/Reseed.cpp
@@ -16,7 +16,7 @@
 #include "util/Log.h"
 #include "Identity.h"
 #include "crypto/CryptoConst.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "util/util.h"
 
 

--- a/core/RouterContext.cpp
+++ b/core/RouterContext.cpp
@@ -6,7 +6,7 @@
 #include "RouterContext.h"
 #include "util/Timestamp.h"
 #include "I2NPProtocol.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "util/util.h"
 #include "version.h"
 

--- a/core/UPnP.cpp
+++ b/core/UPnP.cpp
@@ -17,7 +17,7 @@
 
 #include "RouterContext.h"
 #include "UPnP.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "util/util.h"
 
 #include <miniupnpc/miniupnpc.h>

--- a/core/api.cpp
+++ b/core/api.cpp
@@ -1,7 +1,7 @@
 #include <string>
 #include <map>
 #include "util/Log.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "transport/Transports.h"
 #include "tunnel/Tunnel.h"
 #include "RouterContext.h"

--- a/core/transport/NTCPSession.cpp
+++ b/core/transport/NTCPSession.cpp
@@ -10,7 +10,7 @@
 #include "I2NPProtocol.h"
 #include "RouterContext.h"
 #include "Transports.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "NTCPSession.h"
 
 using namespace i2p::crypto;

--- a/core/transport/SSU.cpp
+++ b/core/transport/SSU.cpp
@@ -3,7 +3,7 @@
 #include "util/Log.h"
 #include "util/Timestamp.h"
 #include "RouterContext.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "SSU.h"
 
 namespace i2p

--- a/core/transport/SSUData.cpp
+++ b/core/transport/SSUData.cpp
@@ -2,7 +2,7 @@
 #include <boost/bind.hpp>
 #include "util/Log.h"
 #include "util/Timestamp.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "SSU.h"
 #include "SSUData.h"
 

--- a/core/transport/Transports.cpp
+++ b/core/transport/Transports.cpp
@@ -3,7 +3,7 @@
 #include "crypto/CryptoConst.h"
 #include "RouterContext.h"
 #include "I2NPProtocol.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "Transports.h"
 
 using namespace i2p::data;

--- a/core/tunnel/Tunnel.cpp
+++ b/core/tunnel/Tunnel.cpp
@@ -9,7 +9,7 @@
 #include "util/Timestamp.h"
 #include "I2NPProtocol.h"
 #include "transport/Transports.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "Tunnel.h"
 
 namespace i2p

--- a/core/tunnel/TunnelEndpoint.cpp
+++ b/core/tunnel/TunnelEndpoint.cpp
@@ -1,7 +1,7 @@
 #include "util/I2PEndian.h"
 #include <string.h>
 #include "util/Log.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "I2NPProtocol.h"
 #include "transport/Transports.h"
 #include "RouterContext.h"

--- a/core/tunnel/TunnelPool.cpp
+++ b/core/tunnel/TunnelPool.cpp
@@ -2,7 +2,7 @@
 #include "util/I2PEndian.h"
 #include "crypto/CryptoConst.h"
 #include "Tunnel.h"
-#include "NetDb.h"
+#include "NetworkDatabase.h"
 #include "util/Timestamp.h"
 #include "Garlic.h"
 #include "transport/Transports.h"

